components/Home/Popular.vue

```diff
@@ -0,0 +1,25 @@
+<template>
+  <div class="mb-20">
+    <div class="flex justify-between items-center mb-11 mt-20">
+      <h2>Most popular</h2>
+      <a href="/fleet">See more -></a>
+    </div>
+    <div class="flex gap-10">
+      <ProductTile
+        v-for="zep in mostPopular"
+        :key="zep.product_id"
+        :zeppelin="zep"
+        class="w-1/4"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { getProductList } from '~/service/products';
+
+const mostPopular = ref();
+onMounted(async () => {
+  mostPopular.value = await getProductList(4);
+});
+</script>
```

Comment:
Is it ok to leave empty `ref()` like here? I got confused at first glance thinking this is a reference to a DOM element 😅 
If we expect it to be a list, so probably an array of elements, wouldn't it be better do add here at least for example `const mostPopular = ref([])`?
Asking out of curiosity, not much experience yet with Composition API 🙏

Reply:
I am not really sure, this is a tip I during bootcamp.
Adding an `ref([])` will cascade down issues with type can not be assigned to type `never|never[]`
This way TS is happy and don't have to import another type to the component.

I might be doing it wrong? Not sure?

Reply:
if there is no exact initializer for `ref` best way is to provide wanted type declaration, then you will may safe pass empty array for init and TS will know what type the ref has 👌 
`const mostPopular = ref<MyProduct[]>([]);`

Reply:
Yeah that's why I asked because I felt like something like `const mostPopular = ref<MyProduct[]>([]);` is missing there. But it was just a guess, still a newbie here and need to warm up much before getting confident in Composition API and TS 😄

Reply:
I think we have this solved because we have a different approach to this now 😄